### ISSUE TITLE
SVGP isn't fitting to data correctly

### ISSUE THREAD

**Issue by connorjardine** [SheffieldML/GPy#793]
Hi, 

I'm a Glasgow University Msci student working on a project to predict fires in Glasgow. 

I have been trying to use GPy to regress the number of fires at each area. 

I have 10553 datapoints.
I've simplified my data down to a y co-ordinate and number of fires:
My X values are in the range 0 to 5844000.
My Y values are in the range 0 to 114.

I'm selecting 800 inducing points using K-means clustering, and reduced the number of iterations to 200.

`Z = cluster_inducing(X, 800)

batchsize = 2000
m = GPy.core.SVGP(X, Y, Z, GPy.kern.RBF(1), GPy.likelihoods.Gaussian(), batchsize=batchsize)

opt = climin.Adadelta(m.optimizer_array, m.stochastic_grad, step_rate=0.2, momentum=0.9)

def callback(i):
    print(str(m.log_likelihood()))
    print(i['n_iter'])
    #Stop after 5000 iterations
    if i['n_iter'] > 200:
        return True
    return False

info = opt.minimize_until(callback)`

This does not work well and is predicting zero for every point in the data.
Basically returning an array of [0, 0, 0, 0 ... , 0, 0, 0] as the picture shows

![gpexample](https://user-images.githubusercontent.com/25176253/68090304-2a3add00-fe6a-11e9-99d9-8212f2d639be.PNG)

I do not think that this is an issue with GPy, but my own use of it. Can you give me any guidance as how to rectify this?

**Comment by mzwiessele** [comment]
There is a few parameters here which you can play around with to fit the data better. One is to normalize your inputs and outputs to live int the range 0-1 before fitting the GP. Second, you have a quite large step size, which might be overshooting the target. Another idea is to initialize the GP without noise variance (setting the likelihood variance of the Gaussian close to zero ~0.001). 

Once you start playing around with the parameters you will see which one, or combination works for your dataset :)
